app/Http/Controllers/AuthController.php

```diff
@@ -4,17 +4,67 @@
 
 use App\Models\Config;
 use App\Models\Store;
+use App\Models\User;
+use App\Services\BigCommerceService;
 use GuzzleHttp\Exception\RequestException;
 use Illuminate\Http\Request;
 use Illuminate\Http\Response;
 
 class AuthController extends Controller
 {
-    public function index()
+    /**
+     * App load url as defined in app configuration
+     */
+    public function load(Request $request, BigCommerceService $bigCommerceService)
     {
-        return new Response('github/plugin-bigcommerce-di');
+        $this->validate($request, ['signed_payload' => 'required']);
+        $signedPayload = $request->input('signed_payload');
+
+        $data = $bigCommerceService->verifySignedRequest($signedPayload);
+        $store = Store::where('context', $data['context'])->first();
+        if (!$store) {
+            return new Response('Error: Store could not be found', 400);
+        }
+
+        $user = User::where('bigcommerce_user_id', $data['user']['id'])->first();
+        if (!$user) {
+            $user = new User();
+            $user->email = $data['user']['email'];
+            $user->role = 'user';
+            $user->bigcommerce_user_id = $data['user']['id'];
+            $user->store_id = $store->id;
+            $user->save();
+        }
+
+        $this->storeToSession([
+            'access_token' => $store['access_token'],
+            'context' => $data['context'],
+            'store_hash' => $data['store_hash']
+        ]);
+
+        $store_id = $store['id'];
+        $configRow = Config::where('store_id', $store_id)->first();
```

Comment:
Lumen is smart enough to figure the name of the row out by its own. You should be able to call `Config::whereStoreId($store_id)`. To not confuse the IDE though simply add the following to the docblock of the entity:

```php
use Illuminate\Database\Eloquent\Builder;

/**
 * @method static Builder|Config whereStoreId(int $storeId)
 */
class Config // extends ...
{
    // ...
}
```